### DISCUSSION TITLE
Are MapCharts working?

### DISCUSSION THREAD

**Discussion by AsenjoPablo** [wuba/react-native-echarts#21] [Q&A] [answered]
Hello,

Are MapCharts functional? I can't find any examples and get "cannot read property 'regions' of undefined", as well as an error when registering the json map on echarts.

Any ideas?
Thank you

**Comment by zhiqingchen** [comment, marked as answer]
what kind of MapCharts you means，Can you give me a link to the echarts example?

**Reply by AsenjoPablo** [reply]
https://echarts.apache.org/examples/en/editor.html?c=map-usa

This would be the kind of example I am using. This is the method I am using for the SVG component:
`function SvgComponent({ option }: { option: any }) {
    const svgRef = useRef<any>(null);

    useEffect(() => {
        let chart: any;
        if (svgRef.current) {
            echarts.registerMap('country', mapJSON)

            // @ts-ignore
            chart = echarts.init(svgRef.current, 'light', {
                renderer: 'svg',
                width: E_WIDTH,
                height: E_HEIGHT,
            });
            chart.setOption(option);
        }
        return () => chart?.dispose();
    }, [option]);

    return <SvgChart ref={svgRef} />;

Map json is imported at the top, but types don't seem to match... which is strange, provided this is working on a NextJS webapp using echarts.

**Reply by zhiqingchen** [reply]
https://github.com/wuba/taro-playground/blob/main/src/pages/explore/charts/pages/map/mapUsaPie.tsx
![image](https://user-images.githubusercontent.com/1876158/219615375-947e3633-9082-4bfa-8574-1de3655355cd.png)

This example can be used as a reference.

**Reply by AsenjoPablo** [reply]
Made a few changes based on the example and it's fully working now. Thank you!
